app/models/manageiq/providers/vmware/infra_manager/vm/operations/snapshot.rb

```diff
@@ -26,9 +27,12 @@ def raw_remove_snapshot(snapshot_id)
 
       run_command_via_parent(:vm_remove_snapshot, :snMor => snapshot.uid_ems)
     rescue => err
-      create_notification(:vm_snapshot_failure, :error => err.to_s, :snapshot_op => "remove")
-      if err.to_s.include?('not found')
```

Comment:
"not found" is not the only exception snapshot removal can hit (and doesn't even catch the "has not been created or has not been completely created" errors), and we shouldn't be raising VimFaults outside the provider